lib/vsc/utils/generaloption.py

```diff
@@ -497,28 +498,38 @@ def is_value_a_commandline_option(self, value, index=None):
 
         The method tests for possible ambiguity on the commandline when the parser
         interprets the argument following an option as a value, whereas it is far more likely that
-        it is (intended as) an option.
+        it is (intended as) an option; --longopt=value is never considered ambiguous, regardless of the value.
         """
         # Values that are/could be options that are passed via
-        # --longopt=value or -sVALUE are not a problem.
+        # --longopt=value is not a problem.
         # When processing the enviroment and/or configfile, we always set
         # --longopt=value, so no issues there either.
 
+        # following checks assume that value is a string (not a store_or_None)
+        if not isinstance(value, basestring):
+            return None
+
+        cmdline_index = None
         try:
             cmdline_index = self.commandline_arguments.index(value)
         except ValueError:
-            # There is no ambiguity if the value is not passed
-            # as standalone argument via commandline
-            return None
+            # no index found for value, so not a stand-alone value
+            if opt.startswith('--'):
+                # --longopt=value is unambigouos
+                return None
 
         if index is None:
             # index of last parsed arg in orig_rargs via remainder of rargs
             index = len(self.commandline_arguments) - len(self.rargs) - 1
 
-        if index != cmdline_index:
+        if cmdline_index is not None and index != cmdline_index:
             # This is not the value you are looking for
             return None
 
+        if not self.ALLOW_OPTION_NAME_AS_VALUE:
+            if '-%s' % value in self._short_opt or '-%s' % value in self._long_opt:
+                return "'-%(val)s' is a valid option, so using '%(val)s' as value is likely a typo" % {'val': value}
```

Comment:
remove the `so using '%(val)s' as value is likely a typo` it's not required (and none of the other error messages have it)